StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1615UnitTests.cs

```diff
@@ -106,14 +133,106 @@ public class ClassName
     /// </summary>
 $$
 }";
-            var expected = new[]
-            {
-                this.CSharpDiagnostic().WithLocation(10, 21)
-            };
+            var fixedCode = @"
+/// <summary>
+/// Foo
+/// </summary>
+public class ClassName
+{
+    /// <summary>
+    /// Foo
+    /// </summary>
+    /// <returns></returns>
+$$
+}";
+
+            var expected = this.CSharpDiagnostic().WithLocation(10, 21);
 
             await this.VerifyCSharpDiagnosticAsync(testCode.Replace("$$", declaration), expected, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedCode.Replace("$$", declaration), EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode.Replace("$$", declaration), fixedCode.Replace("$$", declaration), cancellationToken: CancellationToken.None).ConfigureAwait(false);
         }
-        
+
+        [Theory]
+        [MemberData(nameof(AsynchronousWithReturnValue))]
+        public async Task TestAsynchronousMethodWithReturnTypeWithoutReturnTypeDocumentationAsync(string declaration)
+        {
+            var testCode = @"
+using System.Threading.Tasks;
+/// <summary>
+/// Foo
+/// </summary>
+public class ClassName
+{
+    /// <summary>
+    /// Foo
+    /// </summary>
+$$
+}";
+            var fixedCode = @"
+using System.Threading.Tasks;
+/// <summary>
+/// Foo
+/// </summary>
+public class ClassName
+{
+    /// <summary>
+    /// Foo
+    /// </summary>
+    /// <returns><placeholder>A <see cref=""Task""/> representing the asynchronous operation.</placeholder></returns>
```

Comment:
It's a [SHFB MAML element](http://ewsoftware.github.io/MAMLGuide/html/d995d093-ce18-4a7e-bd03-82c6c0b4fd72.htm) that also works in XML documentation comments. In combination with #536, it plays a vital role in allowing us to generate certain documentation comment (e.g. here and in #606) without compromising the quality of project documentation.